lib/api/backend_medications_api.dart

```diff
@@ -33,75 +33,84 @@ class BackendMedicationsApi extends MedicationsApi {
 
   @override
   Future<Stream<List<Medication>>> getMedications() async {
-    BackendResponse response = await Backend.getAll(Medication.databaseTable);
-
-    if (!response.success && response.error != null) {
-      return Stream<List<Medication>>.error(response.error!);
-    }
-
-    for (BackendObject object in response.results) {
-      _medicationsList.add(
-        Medication(
-          compound: object.get('MedicalProduct'),
-          morning: object.get('Morning').toDouble(),
-          noon: object.get('Noon').toDouble(),
-          evening: object.get('Evening').toDouble(),
-          night: object.get('AtNight').toDouble(),
-          objectId: object.objectId,
-          createdAt: object.createdAt,
-          updatedAt: object.updatedAt,
-        ),
+    try {
+      List<dynamic> response = await Backend.getAll(Medication.databaseTable);
+
+      for (dynamic element in response) {
+        _medicationsList.add(
+          Medication(
+            compound: element['MedicalProduct'],
+            morning: element['Morning'].toDouble(),
+            noon: element['Noon'].toDouble(),
+            evening: element['Evening'].toDouble(),
+            night: element['AtNight'].toDouble(),
+            objectId: element['objectId'],
+            createdAt: DateTime.parse(element['createdAt']),
+            updatedAt: DateTime.parse(element['updatedAt']),
+          ),
+        );
+      }
+
+      return _medicationsController.stream.asBroadcastStream(
+        onListen: (StreamSubscription<List<Medication>> subscription) {
+          _dispatch();
+        },
       );
+    } catch (e) {
+      return Stream<List<Medication>>.error(e);
     }
-
-    return _medicationsController.stream.asBroadcastStream(
-      onListen: (StreamSubscription<List<Medication>> subscription) {
-        _dispatch();
-      },
-    );
   }
 
   @override
   Future<void> saveMedication(Medication medication) async {
-    BackendResponse response = await Backend.saveObject(medication);
-
-    if (!response.success) {
+    try {
+      dynamic response = await Backend.saveObject(medication);
+
+      if (response['updatedAt'] != null) {
+        medication = medication.copyWith(
+          objectId: response['objectId'],
+          updatedAt: DateTime.parse(response['updatedAt']),
+        );
+      }
+
+      if (response['createdAt'] != null) {
+        medication = medication.copyWith(
+          objectId: response['objectId'],
+          createdAt: DateTime.parse(response['createdAt']),
+        );
+      }
```

Comment:
Have a look on 'null-aware operators' in dart, maybe you can have those instead of ifs?